mpu: factorize stm32 mpu support

jamike opened the pull request:
- stm32 mpu configuration files have moved to a common directory to make possible the use for other
stm32 family.
- non regression test on nucleo_f401re : mpu_test and mpu_stack_guard test
-L4 support for all series is added since mpu is supported for all socs from the series (same as for f4)
- nucleo_l476rg and disco_l475_iot1 have passed mpu_test , mpu_stack_guard_test, and the mpu is enabled on these boards.
- F3 mpu support is added only for STM32F373XC (the other socs present in zephyr does support mpu)
- As I don't have board stm32373c_eval, I test the build but I did not push the commit that enables mpu on this board.

galak reviewed (changes requested):
can you provide more comment in the subject of commit message for 'arm: soc: stm32: factorize mpu config'

galak reviewed (changes requested):
Fix commit message for 'boards: nucleo: add MPU for nucleo_l476rg' missing signed-of-by

erwango commented:
In commit messages:
please explain, for each family, if MPU is supported on whole family (should be yes) and then MPU is available or not depending on SoCs (not present on available F3, F1 SoCs as I understood).
Otherwise LGTM

erwango commented on arch/arm/soc/st_stm32/stm32f1/Kconfig.series:
```diff
@@ -12,6 +12,7 @@ config SOC_SERIES_STM32F1X
 	select SOC_FAMILY_STM32
 	select SYS_POWER_LOW_POWER_STATE_SUPPORTED
 	select HAS_STM32CUBE
+	select CPU_HAS_MPU
```
Since MPU is not suported on all SoCs of stm32f1 series, this should be moved (if indeed supported) under Kconfig.defconfig.stm32f10...

jamike replied:
Actually, I did not find a f1 having MPU support

erwango replied:
Hence, we consider CPU_HAS_MPU will be selected
- at series level if we consider all SoCs of a serie support MPU (F4 and L4 for now)
- at SoC level if, inside one series, some SoC support MPU, and some other don't
I'm alined. Though, it'd be nice to store this information somewhere (help section of config STM32_ARM_MPU_ENABLE config flag maybe?)

Otherwise LGTM

erwango commented on arch/arm/soc/st_stm32/stm32f3/Kconfig.series:
```diff
@@ -13,6 +13,7 @@ config SOC_SERIES_STM32F3X
 	select SYS_POWER_LOW_POWER_STATE_SUPPORTED
 	select CPU_HAS_FPU
 	select CPU_HAS_SYSTICK
+	select CPU_HAS_MPU
```
Since MPU is not suported on all SoCs of stm32f series, this should be moved (if indeed supported) under Kconfig.defconfig.stm32f3....

jamike replied:
change performed

jamike committed:
boards: nucleo_l476rg: enable MPU

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

jamike committed:
arm: soc: stm32: make mpu f4 config useable for other family

As other stm32 series support MPU, move common file in a file tree
useable by socs from other series

JIRA: ZEP-2220

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

jamike committed:
arm: soc: stm32: l4: add MPU capability for series

Since all l4 socs support MPU, add MPU capability in KConfig.series

JIRA: ZEP-2220

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

jamike committed:
boards: disco_l475_iot1: enable MPU

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

jamike commented:
I addressed all the above comments. Is something blocking the merge ?

galak commented on arch/arm/soc/st_stm32/common/Makefile:
```diff
@@ -0,0 +1,4 @@
+obj-$(CONFIG_STM32_ARM_MPU_ENABLE) += arm_mpu_regions.o
+
+zephyr: $(KERNEL_HEX_NAME)
```
Why is the hex image support moving here? If you want it common, do it as a separate patch/PR.

galak reviewed (changes requested):
A few minor things:

1. rebase against latest arm branch
2. Remove the HEX image support from this PR.
3. Cleanup commit messages:
* body of commit messages should wrap around 70 char, you seem to break the lines sooner than need be
* remove spaces before ':' in subjects
* add space after ',' in commit body

jamike committed:
arm: soc: stm32: f3: add MPU capability

Since not all socs from f3 series (i.e stm32f334x8 no MPU) have MPU
capability, add capability only for MPU capable socs in Kconfig.soc

JIRA: ZEP-2220

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

galak commented:
can you rebase on latest arm branch

jamike commented:
rebase done on latest arm
